models/migrations/v81.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"code.gitea.io/git"
+	"code.gitea.io/gitea/models"
+	"github.com/go-xorm/xorm"
+)
+
+func fixReleaseSha1OnReleaseTable(x *xorm.Engine) error {
+	type Release struct {
+		ID      int64
+		RepoID  int64
+		Sha1    string
+		TagName string
+	}
+
+	// Update release sha1
+	const batchSize = 100
+	sess := x.NewSession()
+	defer sess.Close()
+
+	var (
+		err          error
+		count        int
+		gitRepoCache = make(map[int64]*git.Repository)
+		repoCache    = make(map[int64]*models.Repository)
+	)
+
+	if err = sess.Begin(); err != nil {
+		return err
+	}
+
+	for start := 0; ; start += batchSize {
+		releases := make([]*Release, 0, batchSize)
+		if err = sess.Limit(batchSize, start).Where("is_tag=?", false).Find(&releases); err != nil {
```

Comment:
You should add order by id otherwise you can get inconsistent results

Reply:
done.